src/Shared/CompilerAssert.cs

```diff
@@ -41,8 +37,8 @@ public override string ToString()
         // override object.Equals
         public override bool Equals(object? obj)
         {
-            return (obj is CompileError)
-              && Equals((CompileError)obj);
+            return (obj is CompileError error)
+              && Equals(error);
```

Comment:
Making this a record strut would eliminate most of the code for `CompileError`

Reply:
Agreed, perhaps even a record struct. I chose not to take on the distraction in spite of the temptation. :)

Reply:
Agreed.... there wasn't enough value in the change so I avoided the distraction.